### ISSUE TITLE
[BUG] General, Data and Appearance are empty

### ISSUE THREAD

**Issue by dbouras** [orblazer/plasma-applet-resources-monitor#2]
First of all, thank you for taking this over - the old resource monitor has always been my favorite!

**To Reproduce**

Right-click on plasmoid -> Configure...

"Configure", "Data" and "Appearance" all are completely empty - I can't configure anything. The only thing that shows something is "Keyboard Shortcuts". The pane doesn't even update when you click a selection.; I discovered this after clicking on "Keyboard Shortcuts" and then on any of the other tabs.

**Versions**

- Version : 2.1.4
- KDE Plasma Version : 5.18.6
- KDE Frameworks Version : 5.71.0
- Qt Version : 5.12.7
- Kernel Version : 5.3.18-lp152.81-preempt
- OS Type : 64-bit
- Graphics Platform : X11

**Comment by orblazer** [comment]
Hello,
That is weird, this has never happen to me, the only issue like that i have see is when i run in `plasmawindowed` for test i have some time all categories stacked.

**Comment by dbouras** [comment]
Is there anything I can do to help find what the issue is?

**Comment by orblazer** [comment]
Do you have try to re install the plasmoid and/or restart your computer ?

**Comment by dbouras** [comment]
Yup, tried all that, and on another computer also. It may have something to do with my user setup. I will investigate further and let you know.

**Comment by ninja85a** [comment]
can confirm as well, I'm running kubuntu 20.04 LTS 
Operating System: Kubuntu 20.04
KDE Plasma Version: 5.18.5
KDE Frameworks Version: 5.68.0
Qt Version: 5.12.8
Kernel Version: 5.11.0-34-generic
OS Type: 64-bit
Processors: 12 × AMD Ryzen 5 1600 Six-Core Processor
Memory: 31.4 GiB of RAM

**Comment by dbouras** [comment]
I found a little bit of time today, debugged the most recent version (v2.2.0) and resolved the issue.

The important first step is to gain visibility into plasma's log. You can do this by restarting plasma from a terminal; for Plasma version > 5.10, this is easily done using:
`killall plasmashell && kstart5 plasmashell`
For other versions have a look here:
https://askubuntu.com/questions/481329/can-i-restart-the-kde-plasma-desktop-without-logging-out

Note that this breaks session management so, when you're done you'll need to logout and login again. But at least you won't need to do this every time you modify and load a plasmoid: when you change something in the source, you just need to restart plasmashell.

From the log it was easy to spot messages such as these:
```
QML StackView: replace: file:///home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigData.qml:2 module "QtQuick.Controls" version 2.15 is not installed
```

This prompted me to review all of your QML code, changing the version for QtQuickControls and Kirigami as per the guide here:
https://doc.qt.io/qt-5/qtquickcontrols-index.html

I adjusted version numbers according to what's shipped with openSuSE Leap 15.2 that comes with Qt 5.12. Then I restarted plasma once again, added the plasmoid on my panel and... voila :) everything works like a charm.

This is my context diff (in case anyone else finds it useful):

```
diff -rc Desktop/org.kde.resourcesMonitor-fork/contents/ui/components/ColorSelector.qml /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/components/ColorSelector.qml
*** Desktop/org.kde.resourcesMonitor-fork/contents/ui/components/ColorSelector.qml  2021-10-20 10:31:56.000000000 -0700
--- /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/components/ColorSelector.qml  2021-10-20 11:42:50.677105547 -0700
***************
*** 1,5 ****
  import QtQuick 2.2
! import QtQuick.Controls 2.15
  import QtQuick.Layouts 1.1
  import QtQuick.Dialogs 1.0

--- 1,5 ----
  import QtQuick 2.2
! import QtQuick.Controls 2.12
  import QtQuick.Layouts 1.1
  import QtQuick.Dialogs 1.0

diff -rc Desktop/org.kde.resourcesMonitor-fork/contents/ui/components/CustomizableSize.qml /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/components/CustomizableSize.qml
*** Desktop/org.kde.resourcesMonitor-fork/contents/ui/components/CustomizableSize.qml   2021-10-20 10:51:24.000000000 -0700
--- /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/components/CustomizableSize.qml   2021-10-20 11:42:54.813150862 -0700
***************
*** 1,5 ****
  import QtQuick 2.2
! import QtQuick.Controls 2.15
  import QtQuick.Layouts 1.1
  import QtQuick.Dialogs 1.0

--- 1,5 ----
  import QtQuick 2.2
! import QtQuick.Controls 2.12
  import QtQuick.Layouts 1.1
  import QtQuick.Dialogs 1.0

diff -rc Desktop/org.kde.resourcesMonitor-fork/contents/ui/components/DoubleSpinBox.qml /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/components/DoubleSpinBox.qml
*** Desktop/org.kde.resourcesMonitor-fork/contents/ui/components/DoubleSpinBox.qml  2021-10-20 10:20:34.000000000 -0700
--- /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/components/DoubleSpinBox.qml  2021-10-20 11:42:58.950405701 -0700
***************
*** 1,5 ****
  import QtQuick 2.2
! import QtQuick.Controls 2.15

  import "./"

--- 1,5 ----
  import QtQuick 2.2
! import QtQuick.Controls 2.12

  import "./"

diff -rc Desktop/org.kde.resourcesMonitor-fork/contents/ui/components/SpinBox.qml /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/components/SpinBox.qml
*** Desktop/org.kde.resourcesMonitor-fork/contents/ui/components/SpinBox.qml    2021-10-20 10:29:40.000000000 -0700
--- /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/components/SpinBox.qml    2021-10-20 11:43:15.206947510 -0700
***************
*** 1,5 ****
! import QtQuick 2.15
! import QtQuick.Controls 2.15

  import "./"

--- 1,5 ----
! import QtQuick 2.12
! import QtQuick.Controls 2.12

  import "./"

diff -rc Desktop/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigAppearance.qml /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigAppearance.qml
*** Desktop/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigAppearance.qml   2021-10-20 11:02:56.000000000 -0700
--- /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigAppearance.qml   2021-10-20 11:39:50.143127758 -0700
***************
*** 1,7 ****
  import QtQuick 2.2
! import QtQuick.Controls 2.15 as QtControls
  import QtQuick.Layouts 1.1 as QtLayouts
! import org.kde.kirigami 2.13 as Kirigami

  import "../components"

--- 1,7 ----
  import QtQuick 2.2
! import QtQuick.Controls 2.12 as QtControls
  import QtQuick.Layouts 1.1 as QtLayouts
! import org.kde.kirigami 2.6 as Kirigami

  import "../components"

diff -rc Desktop/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigData.qml /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigData.qml
*** Desktop/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigData.qml 2021-10-20 11:31:26.000000000 -0700
--- /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigData.qml 2021-10-20 11:39:55.127182361 -0700
***************
*** 1,7 ****
  import QtQuick 2.2
! import QtQuick.Controls 2.15 as QtControls
  import QtQuick.Layouts 1.1 as QtLayouts
! import org.kde.kirigami 2.13 as Kirigami

  import "../components"

--- 1,7 ----
  import QtQuick 2.2
! import QtQuick.Controls 2.12 as QtControls
  import QtQuick.Layouts 1.1 as QtLayouts
! import org.kde.kirigami 2.6 as Kirigami

  import "../components"

diff -rc Desktop/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigGeneral.qml /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigGeneral.qml
*** Desktop/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigGeneral.qml  2021-06-15 11:46:32.000000000 -0700
--- /home/dimitri/.local/share/plasma/plasmoids/org.kde.resourcesMonitor-fork/contents/ui/config/ConfigGeneral.qml  2021-10-20 11:40:03.535274473 -0700
***************
*** 1,7 ****
  import QtQuick 2.2
! import QtQuick.Controls 2.15 as QtControls
  import QtQuick.Layouts 1.1 as QtLayouts
! import org.kde.kirigami 2.13 as Kirigami

  import "../components"

--- 1,7 ----
  import QtQuick 2.2
! import QtQuick.Controls 2.12 as QtControls
  import QtQuick.Layouts 1.1 as QtLayouts
! import org.kde.kirigami 2.6 as Kirigami

  import "../components"
```

 Thank you so much for all the great work!
